_studio/mfx_lib/encode_hw/h264/src/mfx_h264_encode_hw.cpp

```diff
@@ -2000,10 +2000,11 @@ mfxStatus ImplementationAvc::SubmitToMctf(DdiTask * pTask, bool isSceneChange, b
         mfxStatus sts = GetNativeHandleToRawSurface(*m_core, m_video, *pTask, pTask->m_handleRaw);
         if (sts != MFX_ERR_NONE)
             return Error(sts);
-
+        CmSurface2D *
+            cmRawForMCTF = nullptr;
         if(amtMctf->MCTF_Check_Use())
-            pTask->m_cmRawForMCTF = CreateSurface(m_cmDevice, pTask->m_handleRaw, m_currentVaType);
-        MFX_SAFE_CALL(amtMctf->MCTF_PUT_FRAME(pTask->m_cmRawForMCTF, isSceneChange, true, nullptr, doIntraFiltering));
+            cmRawForMCTF = CreateSurface(m_cmDevice, pTask->m_handleRaw, m_currentVaType);
```

Comment:
What does `CreateSurface()` call doing? Is that to create CM wrapper for the surface encoder received as an input or it's actual creation of the new surface?

Reply:
This has been replaced and it is handled directly in side MCTF_PUT_FRAME, no creation of new surface but assignment of input frame into QfIn buffer